liota/dccs/aws_iot.py

```diff
@@ -0,0 +1,180 @@
+
+from liota.dccs.dcc import DataCenterComponent
+from liota.entities.registered_entity import RegisteredEntity
+from liota.entities.edge_systems.edge_system import EdgeSystem
+from liota.entities.devices.device import Device
+from liota.entities.metrics.metric import Metric
+from liota.entities.metrics.registered_metric import RegisteredMetric
+from liota.lib.utilities.si_unit import parse_unit, UnsupportedUnitError
+
+log = logging.getLogger(__name__)
+
+
+class AWSIoT(DataCenterComponent):
+    """
+    DCC for AWSIoT Platform.
+    """
+    def __init__(self, con, enclose_metadata=False):
+        """
+        :param con: DccComms Object
+        :param enclose_metadata: Include Gateway, Device and Metric names as part of payload or not
+        """
+        super(AWSIoT, self).__init__(
+            comms=con
+        )
+        self.enclose_metadata = enclose_metadata
+
+    def register(self, entity_obj):
+        """
+        :param entity_obj: Entity Object
+        :return: RegisteredEntity Object
+        """
+        log.info("Registering resource with AWSIoT DCC {0}".format(entity_obj.name))
+        super(AWSIoT, self).register(entity_obj)
+        if isinstance(entity_obj, Metric):
+            return RegisteredMetric(entity_obj, self, None)
+        else:
+            return RegisteredEntity(entity_obj, self, None)
+
+    def create_relationship(self, reg_entity_parent, reg_entity_child):
+        """
+        This method creates Parent-Child relationship.  Supported relationships are:
+
+               EdgeSystem
+                   |                                      EdgeSystem
+                Device                   (or)                |
+                   |                                    RegisteredMetric
+             RegisteredMetric
+
+        However, A single EdgeSystem can have multiple child Devices and a each Device can have
+        multiple child Metrics.
+
+        :param reg_entity_parent: Registered EdgeSystem or Registered Device Object
+        :param reg_entity_child:  Registered Device or Registered Metric Object
+        :return: None
+        """
+        # Validating here helps in enclose_metadata
+        if not isinstance(reg_entity_parent.ref_entity, EdgeSystem) and \
+                not isinstance(reg_entity_parent.ref_entity, Device):
+            log.error("reg_entity_parent should either be a Registered EdgeSystem or a Device")
+            raise TypeError("reg_entity_parent should either be a Registered EdgeSystem or a Device")
+
+        if not isinstance(reg_entity_child.ref_entity, Device) and \
+                not isinstance(reg_entity_child, RegisteredMetric):
+            log.error("reg_entity_child should either be a Registered Device or Metric")
+            raise TypeError("reg_entity_child should either be a Registered Device or Metric")
+
+        reg_entity_child.parent = reg_entity_parent
+
+    def _get_entity_hierarchy(self, reg_entity):
```

Comment:
Add a comment this might be taken out in utilities if other DCC requires hierarchy.